public/Connect-DbaInstance.ps1

```diff
@@ -645,15 +645,44 @@ function Connect-DbaInstance {
             } elseif ($inputObjectType -eq 'SqlConnection') {
                 $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $inputObject
             } elseif ($inputObjectType -in 'RegisteredServer', 'ConnectionString') {
-                $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $serverName
-                # Parameter TrustServerCertificate changes the connection string be allow connections to instances with the default self-signed certificate
+                $csb = New-Object -TypeName Microsoft.Data.SqlClient.SqlConnectionStringBuilder -ArgumentList $connectionString
+                # I don't see a good way to get the explicitly used keys, so in the meantime I'm using this code.
+                $csbUsedKeys = @( )
+                $csbUsedKeys += ($csb | Format-List | Out-String) -split [Environment]::NewLine | ForEach-Object { if ($_ -match '^Key *: *(.*)$') { $Matches[1] } }
```

Comment:
tricky... some we can't do anything about it but it should be transparent (i.e. "Server" in the connstring, once parsed, gets moved to "Data Source").
But instead of relying on this, to see what keys are specifically used you can use this pseudo-logic

```
foreach($k in $csb.Keys)
{
    
    if ($csb.Contains($k) -and $csb.ShouldSerialize($k))
    {
        $usedKeys += $k
    }
}
```

Reply:
Thanks for "ShouldSerialize" - that does the trick. Contains is always true for all Keys, so as we loop through all the keys we do not need that. Will push a new commit soon.